mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -300,15 +300,21 @@ private[ml] object DefaultParamsWriter {
       paramMap: Option[JValue] = None): String = {
     val uid = instance.uid
     val cls = instance.getClass.getName
-    val params = instance.extractParamMap().toSeq.asInstanceOf[Seq[ParamPair[Any]]]
+    val params = instance.extractParamMap().toSeq
+      .filter(_.param.name != "initialModel").asInstanceOf[Seq[ParamPair[Any]]]
     val jsonParams = paramMap.getOrElse(render(params.map { case ParamPair(p, v) =>
       p.name -> parse(p.jsonEncode(v))
     }.toList))
+    // If the instance has an "initialModel" param and the param is defined, then the initial model
+    // will be saved along with the instance.
+    val initialModelFlag =
+      instance.hasParam("initialModel") && instance.isDefined(instance.getParam("initialModel"))
     val basicMetadata = ("class" -> cls) ~
       ("timestamp" -> System.currentTimeMillis()) ~
       ("sparkVersion" -> sc.version) ~
       ("uid" -> uid) ~
-      ("paramMap" -> jsonParams)
+      ("paramMap" -> jsonParams) ~
+      ("initialModel" -> initialModelFlag)
```

Comment:
I think it would be slightly more robust to decide whether a model being loaded has an `initialModel` by checking for the presence of a directory names "initialModel". I'm not sure how complex that solution would be. We could leave it as TODO or just go with this method. Thoughts?